Examples/E2/Cargo.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace Examples.E2;
+
+public interface ITransfer
+{
+    void Transfer();
+}
+
+public class Air : ITransfer
+{
+    private static Air? instance { get; set; }
+
+    private Air()
+    {
+    }
+
+    public static Air Instance => instance ??= new Air();
+
+    public void Transfer()
+    {
+    }
+}
+
+public class Ship : ITransfer
+{
+    public Ship(string source, string destination)
+    {
+    }
+
+    public void Transfer()
+    {
+    }
+}
+
+public class Train : ITransfer
+{
+    public void Transfer()
+    {
+    }
+}
+
+public abstract class ConcreteTransfer
+{
+    public abstract ITransfer Transfer();
+}
+
+public class ConcreteAir : ConcreteTransfer
+{
+    public override ITransfer Transfer()
```

Comment:
you have a factory here. so manageing the instance of Air is it's responsibility not Air class.